cmd/monaco/integrationtest/v2/settings_integration_test.go

```diff
@@ -287,3 +294,161 @@ func createSettingsClient(t *testing.T, env manifest.EnvironmentDefinition, opts
 	}
 	return dtClient
 }
+
+func TestOrdered_InsertAtFrontWorksWithoutBeingSet(t *testing.T) {
+	const configFolder = "test-resources/settings-ordered/insert-position"
+
+	const manifestFile = configFolder + "/manifest.yaml"
+
+	const specificEnvironment = "platform"
+	const project = "insert-after-not-set"
+	const schema = "builtin:url-based-sampling"
+
+	RunIntegrationWithCleanup(t, configFolder, manifestFile, specificEnvironment, "InsertAfterNotSet", func(fs afero.Fs, tc TestContext) {
+
+		err := monaco.RunWithFSf(fs, "monaco deploy %s --project %s --verbose", manifestFile, project)
+		assert.NoError(t, err)
+		integrationtest.AssertAllConfigsAvailability(t, fs, manifestFile, []string{project}, specificEnvironment, true)
+
+		sClient := createSettingsClientFromManifest(t, fs, manifestFile, "platform")
+		list, err := sClient.List(t.Context(), schema, dtclient.ListSettingsOptions{
+			DiscardValue: true,
+		})
+
+		assert.GreaterOrEqual(t, len(list), 2, "At least the two configs in the test should be deployed")
+
+		first := settingsExternalIdForTest(t, coordinate.Coordinate{Project: project, Type: schema, ConfigId: "first"}, tc)
+		second := settingsExternalIdForTest(t, coordinate.Coordinate{Project: project, Type: schema, ConfigId: "second"}, tc)
+
+		assert.Equal(t, len(list)-2, findPositionWithExternalId(t, list, first))
```

Comment:
Should we maybe add `Greater` instead (also for other tests)? This would then less interfere with other E2E in parallel.
```go
assert.Greater(t, findPositionWithExternalId(t, list, second), findPositionWithExternalId(t, list, first))
```

Reply:
I am not a fan of that, as these asserts here are the core of the test. I expect them to be precisely where they are.